config/initializers/csv_bucket.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+if Rails.application.config.x.vcap_services.present?
+
+  if ENV["BUCKET_NAME"].blank?
+    bucket_name = Rails.application.config.x.vcap_services.dig(
+      "aws-s3-bucket", 0, "credentials", "bucket_name"
```

Comment:
This seems fragile. I think we want to pick the one with the correct name, instead of taking the first. What happens when we need a second bucket?

Reply:
Could we not just change the name of the second service in terraform, or would that not work? Does every bucket service have to be "aws-s3-bucket" rather than "01_aws-s3-bucket" etc?

Reply:
```
irb(main):017:0> JSON.parse(ENV["VCAP_SERVICES"])["aws-s3-bucket"].select{ |bucket| bucket["name"] = "ecf-csv-dev"}
=> [{"label"=>"aws-s3-bucket", "provider"=>nil, "plan"=>"default", "name"=>"ecf-csv-dev", "tags"=>["s3"], "instance_guid"=>"0db3af2d-f7af-4d73-9bf6-28c2dfa550a7", "instance_name"=>"ecf-csv-dev", "binding_guid"=>"e852f53e-2bd8-4e75-b45a-03815fbe98c4", "binding_name"=>nil, "credentials"=>{"bucket_name"=>"paas-s3-broker-prod-lon-0db3af2d-f7af-4d73-9bf6-28c2dfa550a7", "aws_access_key_id"=>"AKIAV3ON3AJYLIATVON5", "aws_secret_access_key"=>"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", "aws_region"=>"eu-west-2", "deploy_env"=>""}, "syslog_drain_url"=>nil, "volume_mounts"=>[]}]
irb(main):018:0> Rails.env
=> "deployed_development"
irb(main):019:0> JSON.parse(ENV["VCAP_SERVICES"])["aws-s3-bucket"].select{ |bucket| bucket["name"].include?("csv")}
=> [{"label"=>"aws-s3-bucket", "provider"=>nil, "plan"=>"default", "name"=>"ecf-csv-dev", "tags"=>["s3"], "instance_guid"=>"0db3af2d-f7af-4d73-9bf6-28c2dfa550a7", "instance_name"=>"ecf-csv-dev", "binding_guid"=>"e852f53e-2bd8-4e75-b45a-03815fbe98c4", "binding_name"=>nil, "credentials"=>{"bucket_name"=>"paas-s3-broker-prod-lon-0db3af2d-f7af-4d73-9bf6-28c2dfa550a7", "aws_access_key_id"=>"AKIAV3ON3AJYLIATVON5", "aws_secret_access_key"=>"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", "aws_region"=>"eu-west-2", "deploy_env"=>""}, "syslog_drain_url"=>nil, "volume_mounts"=>[]}]
```

Reply:
how about: `JSON.parse(ENV["VCAP_SERVICES"])["aws-s3-bucket"].select{ |bucket| bucket["name"].include?("csv")}` ?